tests/generation/test_framework_agnostic.py

```diff
@@ -0,0 +1,40 @@
+"""
+Framework agnostic tests for generate()-related methods.
```

Comment:
The goal would be to expand this file with other generation integration tests, logits processors tests, and model-inheritable generation tests